src/metabase/sync/sync_metadata/indexes.clj

```diff
@@ -13,27 +13,47 @@
    :added-indexes 0
    :removed-indexes 0})
 
+(defn- nested-field-names->field-id
```

Comment:
There's a column on `metabase_field` called `nfc_path` that should have the information you need to implement this function without resorting to recursion. I think this should be a function inside the `field` namespace though.

Reply:
Oh sorry, that's for JSON fields in SQL databases. I'm surprised something similar doesn't exist for mongoDB, oh well

Reply:
moved in https://github.com/metabase/metabase/pull/36624/commits/f96c83c58fd4e479576d6d22b41e6a3ac01273f9